### PR TITLE
(feat: coalescing) added some docs.

### PR DESCRIPTION
**Pull request by hoebbelsB**
added a simple documentation for the coalesce pipe

### CODE REVIEW

File: libs/rxjs-state/src/lib/core/operators/coalesce.md [hunk 1]
```diff
@@ -1,15 +1,78 @@
...
- ## Example
+defaultCoalesceDurationSelector = <T>(value: T) => generateFrames();
 
- ```ts
+defaultCoalesceConfig: CoalesceConfig = {
+      context: {isCoalescing: false},
+      leading: false,
+      trailing: true
+};
+  ```
```
Comment (BioPhoton) [inline libs/rxjs-state/src/lib/core/operators/coalesce.md]: The headline is usage the content looks more like signature and defaults?

File: libs/rxjs-state/src/lib/core/operators/coalesce.md [hunk 2]
```diff
@@ -1,15 +1,78 @@
...
+ source$.pipe(
+     coalesce(() => generateFrames(), coalesceConfig)
+ ).subscribe(coalescedVal => {
+     console.log(coalescedVal);
+ });
+ // output: 4
+ source$.pipe(
+     coalesce(() => generateFrames(), coalesceConfig)
+ ).subscribe(coalescedVal => {
+     console.log(coalescedVal);
```
Comment (BioPhoton) [inline libs/rxjs-state/src/lib/core/operators/coalesce.md]: Maybe put some more meaningful like a render() function?

File: libs/rxjs-state/src/lib/core/operators/coalesce.md [hunk 3]
```diff
@@ -1,15 +1,78 @@
...
+ });
+// no output, since the value will be emitted only once per scope
+   ```
```
Comment (BioPhoton) [inline libs/rxjs-state/src/lib/core/operators/coalesce.md]: Imho more examples that show some osecases would be a good idea.

Or at least show the different configurations separately.

File: libs/rxjs-state/src/lib/core/operators/coalesce.md [hunk 4]
```diff
@@ -1,15 +1,78 @@
- Should emit once per micro task across usage.
+ Emits a value once per animationFrame, then ignores subsequent source
```
[the remark this answers is not in the document]
Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/coalesce.md]: see https://github.com/BioPhoton/rxjs-state/pull/24/commits/a7dd5781dd06c7b07cbde30165d9fdd2a7b75871

File: libs/rxjs-state/src/lib/core/operators/coalesce.md (hunk 1 above)
Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/coalesce.md]: adapted the structure. see https://github.com/BioPhoton/rxjs-state/pull/24/commits/6447e0d011ec4013103fd64346a61585aa492101

Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/coalesce.md]: and here https://github.com/BioPhoton/rxjs-state/pull/24/commits/d3c9f116a8488ae319c41855c39e0c60401f4945

File: libs/rxjs-state/src/lib/core/operators/coalesce.md (hunk 2 above)
Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/coalesce.md]: yeah, put a bit more effort in describing what happens and what it is used for. see https://github.com/BioPhoton/rxjs-state/pull/24/commits/d3c9f116a8488ae319c41855c39e0c60401f4945

Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/coalesce.md]: put some more love to it...

File: libs/rxjs-state/src/lib/core/operators/coalesce.md [hunk 5]
```diff
@@ -1,15 +1,78 @@
- Should emit once per micro task across usage.
+ Emits a value once per animationFrame, then ignores subsequent source 
+ values for the duration of one animationFrame, then repeats this process. 
 
  ![](coalesce.png)
 
- Detailed description
+ # Description
+ Rendering, in most web applications, is by far the most performance crucial part.
+ The _coalesce_ operator's general purpose is to buffer render critical state changes together
```
[the remark this answers is not in the document]
Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/coalesce.md]: yes, described it in a more clear way. see https://github.com/BioPhoton/rxjs-state/pull/24/commits/e062c8bd218fd9a79c3d5e0b7a9ec12671facc88

File: libs/rxjs-state/src/lib/core/operators/coalesce.md [hunk 6]
```diff
@@ -1,15 +1,78 @@
...
- Detailed description
+ # Description
+ Rendering, in most web applications, is by far the most performance crucial part.
+ The _coalesce_ operator's general purpose is to buffer render critical state changes together 
+ to a single emission. By default changes will be emitted on the trailing end of an animationFrame 
+ but this behavior is fully configurable.
+ 
+ Furthermore the _coalesce_ operator provides the option to define a custom scope. If provided, 
+ changes will only be emitted once per scope. This helps especially in component based 
+ architectures.
```
[the remark this answers is not in the document]
Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/coalesce.md]: yes, here is a more general explanation approach: https://github.com/BioPhoton/rxjs-state/pull/24/commits/119901ac1983ce0a9491cfadaf5244d7c369e19d

**Pull request merged**